HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -2729,86 +2699,56 @@ def self.convert_curve_biquadratic(coeff)
-    end
-    if not max_y.nil?
-      curve.setMaximumCurveOutput(max_y)
+      for var_value in (var[:sample_low]...var[:sample_high]).step(var[:sample_step])
+        var[:values] << var_value
+      end
     end
-    if is_dimensionless
-      curve.setInputUnitTypeforX('Dimensionless')
-      curve.setOutputUnitType('Dimensionless')
+    curve_output_values = []
+    if vars.size == 2
+      vars[0][:values].each do |var1_value|
+        vars[1][:values].each do |var2_value|
+          if curve_type == 'biquadratic'
+            curve_output_values << MathTools.biquadratic(var1_value, var2_value, curve_coeff)
+          elsif curve_type == 'bicubic'
+            curve_output_values << MathTools.bicubic(var1_value, var2_value, curve_coeff)
+          end
+        end
+      end
+    elsif vars.size == 1
+      vars[0][:values].each do |var1_value|
+        if curve_type == 'quadratic'
+          curve_output_values << MathTools.quadratic(var1_value, curve_coeff)
+        end
+      end
     end
-    return curve
+    return vars, curve_output_values
   end
 
-  def self.create_curve_cubic(model, coeff, name, min_x, max_x, min_y, max_y)
-    curve = OpenStudio::Model::CurveCubic.new(model)
-    curve.setName(name)
-    curve.setCoefficient1Constant(coeff[0])
-    curve.setCoefficient2x(coeff[1])
-    curve.setCoefficient3xPOW2(coeff[2])
-    curve.setCoefficient4xPOW3(coeff[3])
-    curve.setMinimumValueofx(min_x)
-    curve.setMaximumValueofx(max_x)
-    curve.setMinimumCurveOutput(min_y)
-    curve.setMaximumCurveOutput(max_y)
-    return curve
+  def self.create_table_lookup_constant(model)
+    var_constant = { name: 'constant_var', min: -100, max: 100, values: [0, 1] }
+    return create_table_lookup(model, 'ConstantTable', [var_constant], [1, 1], -100, 100)
   end
 
-  def self.create_curve_exponent(model, coeff, name, min_x, max_x)
-    curve = OpenStudio::Model::CurveExponent.new(model)
-    curve.setName(name)
-    curve.setCoefficient1Constant(coeff[0])
-    curve.setCoefficient2Constant(coeff[1])
-    curve.setCoefficient3Constant(coeff[2])
-    curve.setMinimumValueofx(min_x)
-    curve.setMaximumValueofx(max_x)
-    return curve
+  def self.create_table_lookup(model, name, independent_vars, output_values, output_min=nil, output_max=nil)
+    table = OpenStudio::Model::TableLookup.new(model)
+    table.setName(name)
+    independent_vars.each do |var|
+      ind_var = OpenStudio::Model::TableIndependentVariable.new(model)
+      ind_var.setName(var[:name])
+      ind_var.setMinimumValue(var[:min])
+      ind_var.setMaximumValue(var[:max])
+      ind_var.setExtrapolationMethod('Linear')
+      ind_var.setValues(var[:values])
+      table.addIndependentVariable(ind_var)
+    end
+    table.setMinimumOutput(output_min) unless output_min.nil?
+    table.setMaximumOutput(output_max) unless output_max.nil?
+    table.setOutputValues(output_values)
+    return table
   end
 
   def self.create_curve_quad_linear(model, coeff, name)
```

Comment:
For GSHP curves I didn't move forward to replace them because they need to create 4 or 5 independent variables for each curve, I am not sure if it's an efficient path to go.